app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt

```diff
@@ -189,39 +171,66 @@ private fun ImportFilesScreen(
         content = {
             Column(
                 modifier = Modifier
-                    .padding(horizontal = Margin.Medium, vertical = Margin.Large)
+                    .widthIn(max = Dimens.MaxSinglePaneScreenWidth)
+                    .padding(vertical = Margin.Small)
                     .verticalScroll(rememberScrollState()),
             ) {
-                ImportFilesTitle(titleRes = R.string.myFilesTitle)
-                ImportedFilesCard(
-                    modifier = Modifier.padding(vertical = Margin.Medium),
-                    files = files,
-                    humanReadableSize = { humanReadableSize },
-                    pickFiles = ::pickFiles,
-                    removeFileByUid = removeFileByUid,
-                )
-                ImportTextFields(transferMessage, selectedTransferType.get)
-                ImportFilesTitle(Modifier.padding(vertical = Margin.Medium), titleRes = R.string.transferTypeTitle)
-                TransferTypeButtons(selectedTransferType)
-                ImportFilesTitle(Modifier.padding(vertical = Margin.Medium), titleRes = R.string.advancedSettingsTitle)
-                TransferOptionsTypes(
-                    transferOptionsStates = transferOptionsCallbacks.transferOptionsStates,
-                    onClick = { selectedOptionType -> showTransferOption = selectedOptionType },
+                FilesToImport(files, humanReadableSize, removeFileByUid, addFiles, shouldStartByPromptingUserForFiles)
+                ImportTextFields(
+                    modifier = Modifier
+                        .padding(horizontal = Margin.Medium)
+                        .fillMaxWidth(),
+                    transferMessage = transferMessage,
+                    selectedTransferType = selectedTransferType.get,
                 )
+                SendByOptions(selectedTransferType)
+                TransferOptions(transferOptionsCallbacks)
             }
-
-            TransferOptions({ showTransferOption }, transferOptionsCallbacks, ::closeTransferOption)
         }
     )
 }
 
 @Composable
-private fun ColumnScope.ImportTextFields(transferMessage: GetSetCallbacks<String>, selectedTransferType: () -> TransferTypeUi) {
+private fun FilesToImport(
+    files: () -> List<FileUi>,
+    humanReadableSize: String,
+    removeFileByUid: (uid: String) -> Unit,
+    addFiles: (List<Uri>) -> Unit,
+    shouldStartByPromptingUserForFiles: Boolean,
+) {
+    var shouldShowInitialFilePick by rememberSaveable { mutableStateOf(shouldStartByPromptingUserForFiles) }
 
-    EmailAddressesTextFields(selectedTransferType)
+    val filePickerLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.OpenMultipleDocuments(),
+        onResult = addFiles,
+    )
 
+    fun pickFiles() {
+        shouldShowInitialFilePick = false
+        filePickerLauncher.launch(arrayOf("*/*"))
+    }
+
+    LaunchedEffect(Unit) { if (shouldShowInitialFilePick) pickFiles() }
```

Comment:
Having this logic be defined here somewhat feels unnatural yet this is the only place where it is needed so I'm not sure what to think of it

Reply:
We decided to keep it here to avoid overhead code